### ISSUE TITLE
Do you remove some records from the test set?

### ISSUE THREAD

**Issue by Spico197** [HangYang-NLP/DE-PPN#3]
Hi there, thanks for your work.

From the limit information of your codes, [num_roles](https://github.com/HangYang-NLP/DE-PPN/blob/2ae062dbd7e39328a1d6d9e6ab1160e266295477/DEE/setpred4DEE.py#L108) is fixed for all the records in `doc_span_info`.
However there are some documents with more than one event type.
**Q1:** Do you remove these documents from the test set and report the final results in your paper?

**Q2:** We can see that you override the `doc_fea.generate_dag_info_for` method, How do you build the golden targets please?

Besides, from the [get_eval_on_doc](https://github.com/HangYang-NLP/DE-PPN/blob/2ae062dbd7e39328a1d6d9e6ab1160e266295477/DEE/dee_model.py#L452) function in the main model, the format of returning values are not same (line 440, 452).
**Q3:** Do you change the evaluation part, or are the final results comparable with the results from the Doc2EDAG paper?

I would very appreciate if you could release the whole project codes and provide a bash entry script to show the hyper-parameters setting.
Thanks for your work and looking forward to your reply.

**Comment by Spico197** [comment]
Another bug: [get_eval_on_doc](https://github.com/HangYang-NLP/DE-PPN/blob/2ae062dbd7e39328a1d6d9e6ab1160e266295477/DEE/dee_model.py#L444) function get the event classification loss while evaluating.

*Q4:* Do you train on the dev and test set ? Otherwise, the returned `event_pred_list` is a tuple and the latter [condition statement](https://github.com/HangYang-NLP/DE-PPN/blob/2ae062dbd7e39328a1d6d9e6ab1160e266295477/DEE/dee_model.py#L446) makes no sense.

**Comment by HangYang-NLP** [comment]
Thank you for your advice, your questions are answered below.
In the procession of training and evaluation, for each predicted event type, our model initial different num_roles, so the num_roles is not fixed. 
For Q1, we don't remove the documents with multiple events from the test set. 
For Q2, the generate_dag_infor_for has been uploaded.
For Q3, in the procession of training and evaluation for each epoch, we set a maximum limit on the number of golden events.
This step is to keep the number of events generated consistent with the number of gold events for matching loss computation.
But for the final evaluation, we evaluate our method as the same evaluation standard as Doc2EDAG.
For Q4, we don't train on the dev or test set. I'm puzzled by this question.
